src/components/GameBoard.js

```diff
@@ -0,0 +1,105 @@
+import React, {useState} from "react";
+import Square from "./Square";
+import GameModeForm from "./GameModeForm";
+
+const GameBoard = ({endGame}) => {
+    const [gameBoard,setGameBoard] = useState([[]])
+    const [flagsCount, setFlagsCount] = useState(-1);
+    const [matrixSize, setMatrixSize] = useState({rows:0,cols:0});
+
+    const decrementFlagsCount = () =>{setFlagsCount(flagsCount-1)}
+    const incrementFlagsCount = () =>{ setFlagsCount(flagsCount+1)}
+
+
+    const gameModeLogic = (gameMode) =>{
+        setGameBoard([[]]);
+        setMatrixSize({rows:0,cols:0});
+        switch(gameMode){
+            case "none":
+                break;
+            case "beginner":
+                setMatrixSize({rows:9,cols:9})
+                createGameMatrix(9,9,10);
+                break;
+            case "intermediate":
+                setMatrixSize({rows:16,cols:16})
+                createGameMatrix(16,16,40)
+                break;
+            case "expert":
+                setMatrixSize({rows:16,cols:30})
+                createGameMatrix(16,30,99)
+                break;
+            default:
+                break;
+        }
+    }
+    const createGameMatrix = (rows,cols,mines) =>{
+        setFlagsCount(mines);
+        const mineShuffle = [];
+        createShuffledMines(mineShuffle,rows,cols,mines);
+        setGameBoard([[]])
+        for (let i = 0; i < rows; i++) {
+            let row = [];
+            for (let j = 0; j < cols; j++) {
+                row.push( mineShuffle.pop());
+            }
+            setGameBoard(prevState => [...prevState, row]);
+        }
+        console.log(gameBoard);
+
+    }
+    const createShuffledMines = (mineShuffle, rows, cols, mines) => {
+        for (let i = 0; i < mines; i++) {
+            mineShuffle.push(true);
+        }
+        for (let i = 0; i < rows * cols - mines; i++) {
+            mineShuffle.push(false);
+        }
+        for (let i = mineShuffle.length - 1; i > 0; i--) {
+            const randomCol = Math.floor(Math.random() * (i + 1));
+            [mineShuffle[i], mineShuffle[randomCol]] = [mineShuffle[randomCol], mineShuffle[i]];
+        }
+    };
+    const calculateValue = (row, col) => {
```

Comment:
what does this function do? what is Value? feel too general for me
if the function calculates how many mines surround the cell at `row` and `col`.
Then perhaps you can rename it to `calculateSurroundMines` or somethings else that represent the meaning of the function more specifically